pkg/client/factory.go

```diff
@@ -165,6 +177,27 @@ func WithPollingConfigManagerDatafileAccessToken(pollingInterval time.Duration,
 	}
 }
 
+// WithOptimizelySdkSettings sets optimizelySdkSettings on a client.
+func WithOptimizelySdkSettings(optimizelySdkSettings *OptimizelySdkSettings) OptionFunc {
+	return func(f *OptimizelyFactory) {
+		f.optimizelySDKSettings = optimizelySdkSettings
+	}
+}
+
+// WithOdpManager sets odp manager on a client.
+func WithOdpManager(odpManager odp.Manager) OptionFunc {
+	return func(f *OptimizelyFactory) {
+		f.odpManager = odpManager
+	}
+}
```

Comment:
It looks like we use combinations of config files (cacheSize, cacheTimeout, enabled) and builder method for ODP configuration. Can we consider removing config files and adding them ODP Builder (withSegmentCacheSize, withSegmentCacheTimeout) just like other ODP configuration?